orchagent/portsorch.cpp

```diff
@@ -211,6 +211,24 @@ PortsOrch::PortsOrch(DBConnector *db, vector<table_name_with_pri_t> &tableNames)
         SWSS_LOG_ERROR("Watermark flex counter groups were not set successfully: %s", e.what());
     }
 
+    string portRatePluginName = "port_rates.lua";
+
+    try
+    {
+        string portRateLuaScript = swss::loadLuaScript(portRatePluginName);
+        string portRateSha = swss::loadRedisScript(m_counter_db.get(), portRateLuaScript);
+
+        vector<FieldValueTuple> fieldValues;
+        fieldValues.emplace_back(PORT_PLUGIN_FIELD, portRateSha);
+        fieldValues.emplace_back(POLL_INTERVAL_FIELD, PORT_RATE_FLEX_COUNTER_POLLING_INTERVAL_MS);
+        fieldValues.emplace_back(STATS_MODE_FIELD, STATS_MODE_READ);
+        m_flexCounterGroupTable->set(PORT_STAT_COUNTER_FLEX_COUNTER_GROUP, fieldValues);
+    }
+    catch (const runtime_error &e)
+    {
+        SWSS_LOG_WARN("Port rate flex counter group plugins was not set successfully: %s", e.what());
+    }
```

Comment:
duplicated code

Reply:
Combined into one try block

Reply:
Sorry for misleading. I mean the code change is almost the same as in `IntfsOrch::IntfsOrch()`. Is it possible to reuse code?

---
In reply to: [412570352](https://github.com/Azure/sonic-swss/pull/1201#discussion_r412570352) [](ancestors = 412570352)